src/Classes/GemSelectControl.lua

```diff
@@ -88,49 +90,73 @@ function GemSelectClass:FilterSupport(gemId, gemData)
 end
 
 function GemSelectClass:BuildList(buf)
+	local searchTerm = ""
+	local tagsList = {}
+
 	self.controls.scrollBar.offset = 0
 	wipeTable(self.list)
 	self.searchStr = buf
-	if self.searchStr:match("%S") then
-		-- Search for gem name using increasingly broad search patterns
-		local patternList = {
-			"^ "..self.searchStr:lower().."$", -- Exact match
-			"^"..self.searchStr:lower():gsub("%a", " %0%%l+").."$", -- Simple abbreviation ("CtF" -> "Cold to Fire")
-			"^ "..self.searchStr:lower(), -- Starts with
-			self.searchStr:lower(), -- Contains
-		}
+	-- if self.searchStr:match("%S") then
+	if #self.searchStr > 0 then
 		local added = { }
-		for i, pattern in ipairs(patternList) do
-			local matchList = { }
-			for gemId, gemData in pairs(self.gems) do
-				if self:FilterSupport(gemId, gemData) and not added[gemId] and ((" "..gemData.name:lower()):match(pattern) or altQualMap[self:GetQualityType(gemId)]:lower():match(pattern)) then
-					t_insert(matchList, gemId)
-					added[gemId] = true
+
+		-- split the buffer using :
+		-- Remove the first entry as the name search term
+		tagsList = self.searchStr:split(':')
+		searchTerm = tagsList[1]
+		t_remove(tagsList,1)
+
+		if searchTerm then
+			-- Search for gem name using increasingly broad search patterns
+			local patternList = {
+				"^ "..searchTerm:lower().."$", -- Exact match
+				"^"..searchTerm:lower():gsub("%a", " %0%%l+").."$", -- Simple abbreviation ("CtF" -> "Cold to Fire")
+				"^ "..searchTerm:lower(), -- Starts with
+				searchTerm:lower(), -- Contains
+			}
+			for i, pattern in ipairs(patternList) do
+				local matchList = { }
+				for gemId, gemData in pairs(self.gems) do
+					if self:FilterSupport(gemId, gemData) and not added[gemId] and ((" "..gemData.name:lower()):match(pattern) or altQualMap[self:GetQualityType(gemId)]:lower():match(pattern)) then
+						addThisGem = true
+						if #tagsList > 0 then
+							for i, v in ipairs(tagsList) do
+								local tagName = string.gsub(v, "%s+", ""):lower()
+								local negateTag = string.sub(tagName, 1, 1) == '-'
```

Comment:
style

```suggestion
							for _, tag in ipairs(tagsList) do
								local tagName = tag:gsub("%s+", ""):lower()
								local negateTag = tagName:sub(1, 1) == '-'
```

Reply:
![image](https://user-images.githubusercontent.com/4302241/150661230-491f0a91-9abe-4caa-bbb5-19304dfc46fa.png)
I wonder if that was why i had it the way it was.

Reply:
> ![image](https://user-images.githubusercontent.com/4302241/150661230-491f0a91-9abe-4caa-bbb5-19304dfc46fa.png) I wonder if that was why i had it the way it was.

Nah, it's because you need to get rid of the first parameter when calling `gsub` this way.  `string.gsub` takes the starting string as a first parameter, but when calling the function directly from the starting string you don't need it